solutions/4.js

```diff
@@ -0,0 +1,9 @@
+const solution = (str, num, i = 0, newStr = '') => {
+  if(i == num){
+    return newStr;
+  }
+  newStr = str + ' ' + newStr;
+  return solution(str, num, i + 1, newStr);
```

Comment:
you are creating unnecessary variables. `return solution(str, num, i+1, str ${newStr})`